xwiki-rendering-api/src/main/java/org/xwiki/rendering/transformation/IconProvider.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.transformation;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.block.IconBlock;
+import org.xwiki.stability.Unstable;
+
+
+/**
+ * Component to provide an icon from its name.
+ *
+ * @version $Id$
+ * @since 15.10.4
+ */
+@Role
+@Unstable
+public interface IconProvider
+{
+    /**
+     * Provides an image icon from its name.
+     * @param iconName the name of the icon needed
+     * @return the block containing the icon
+     */
+    IconBlock get(String iconName);
```

Comment:
This would open the door to implementing iconProviders that would break the icon transformation feature. I think it's for the best to rely on a more precise class with minimal expectations.

Reply:
I have the feeling you kind of misunderstood my original idea. My idea was that the icon transformation wraps whatever block the `IconProvider` returns inside an `IconBlock`, but an `IconBlock` class that is private to the icon transformation. That way, the icon transformation knows also that this `IconBlock` has exactly one child. This `IconBlock` class would not be used anywhere else.